pytorch_lightning/trainer/trainer.py

```diff
@@ -1189,6 +1180,21 @@ def test(self, model: Optional[LightningModule] = None):
         self.run_evaluation(test_mode=True)
 
 
+class _PatchDataLoader(object):
+    r'''
+    Callable object for patching dataloaders passed into trainer.fit().
+    Use this class to override model.*_dataloader() and be pickle-compatible.
+
+    Args:
+        dataloader: Dataloader object to return when called.
+    '''
+    def __init__(self, dataloader):
```

Comment:
```suggestion
    def __init__(self, dataloader: DataLoader):
```

Reply:
For the val and test cases, could we receive a list of dataloaders?  the variable names are plural: `val_dataloaders` and `test_dataloaders`.  Are they lists of loaders by this point?

In that case it should be 
`-> Union[list, DataLoader]` correct?

Reply:
true so rather `-> Union[List[DataLoader], DataLoader]`